### PR TITLE
UPSTREAM: 47281: Update devicepath with filepath.Glob result

### PR DESCRIPTION
**Pull request by mtanino**
If iscsiTransport is not tcp, iSCSI plugin tries to
find devicepath using filepath.Glob but never updates
devicepath with the filepath.Glob result.

This patch fixes the problem.

### CODE REVIEW

**Comment by mtanino** [comment]
@rootfs @humblec @childsb @eparis

**Comment by rootfs** [comment]
lgtm
[test]

**Comment by mtanino** [comment]
[test]

**Comment by mtanino** [comment]
[test]
Network related test failed. not related to this fix.
```
[ERROR] PID 23302: test/extended/networking-minimal.sh:6: `NETWORKING_E2E_MINIMAL=1 "${OS_ROOT}/test/extended/networking.sh"` exited with status 1.
```

**Comment by mtanino** [comment]
@rootfs 
Could you kick test again? Seems I can't do it.

**Comment by mtanino** [comment]
ping,
@childsb @eparis

**Comment by deads2k** [comment]
[test]

@rootfs

**Comment by rootfs** [comment]
[merge]

**Comment by rootfs** [comment]
@childsb  can you merge?

**Comment by eparis** [comment]
[merge]

File: vendor/k8s.io/kubernetes/pkg/volume/iscsi/iscsi_util.go [hunk 1]
```diff
@@ -90,32 +90,39 @@ func updateISCSINode(b iscsiDiskMounter, tp string) error {
...
+					// There might be a case that fpath contains multiple device paths if
+					// multiple PCI devices connect to same iscsi target. We handle this
+					// case at subsequent logic. Pick up only first path here.
+					*devicePath = fpath[0]
+				}
 			}
+			if err == nil {
+				return true
+			}
+			if err != nil && !os.IsNotExist(err) {
```
Comment (eparis) [inline vendor/k8s.io/kubernetes/pkg/volume/iscsi/iscsi_util.go]: How did no one point out this `err != nil` is redundant? Don't have to fix it here, but we should fix this upstream.

Comment (mtanino) [inline vendor/k8s.io/kubernetes/pkg/volume/iscsi/iscsi_util.go]: I'll push to remove that redundant "error != nil" check to upstream.

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[severity:bug][merge]

**Pull request merged**